sklearn/feature_extraction/text.py

```diff
@@ -284,6 +284,18 @@ def build_analyzer(self):
             stop_words = self.get_stop_words()
             tokenize = self.build_tokenizer()
 
+            inconsistent = set()
+            for w in stop_words or ():
+                tokens = list(tokenize(preprocess(w)))
+                for token in tokens:
+                    if token not in stop_words:
+                        inconsistent.add(token)
+            if inconsistent:
```

Comment:
I wonder if it should be cached somehow, as build_analyzer is called in transform method, so it may mean a ~100 additional words to process per each .transform call.

Reply:
Maybe,
```py
def _check_stop_words_consistency(self, stop_words):

    if not (hasattr(self, _stop_words_id) or id(stop_words) != self._stop_words_id):
       inconsistent = set()
       for w in stop_words or ():
           tokens = list(tokenize(preprocess(w)))
           for token in tokens:
               if token not in stop_words:
                   inconsistent.add(token)
       self._stop_words_inconsistent = inconsistent
       self._stop_words_id = id(stop_words)

    if self._stop_words_inconsistent:
         warnings.warn('Your stop_words may be inconsistent with your '
                       'preprocessing. Tokenizing the stop words '
                       'generated tokens %r not in stop_words.' %
                       sorted(self._stop_words_inconsistent))
```

as I think using e.g. `joblib.Memory`  will pickle the stop word list to check for object equality which will might also have a significant overhead.

Though this would add 2 private attributes..

Reply:
Seems reasonable, @glemaitre.